### PR TITLE
Add pytest and datasets to dev dependencies

### PR DESCRIPTION
**Pull request by chiwanpark**
## Summary
To run the convergence and correctness tests, we need to `pytest` and `datasets` packages. This PR adds the packages to the dev dependencies.

## Testing Done

- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

```zsh
❯ make test
pytest --disable-warnings test/ --ignore=test/convergence
================================================ test session starts =================================================
platform linux -- Python 3.10.12, pytest-8.3.2, pluggy-1.5.0
rootdir: /data/ovis_915/bell/Liger-Kernel.git
plugins: anyio-4.4.0
collected 114 items                                                                                                  

test/transformers/test_cross_entropy.py ..........................................................             [ 50%]
test/transformers/test_fused_linear_cross_entropy.py ......                                                    [ 56%]
test/transformers/test_geglu.py ........                                                                       [ 63%]
test/transformers/test_rms_norm.py ................                                                            [ 77%]
test/transformers/test_rope.py ............                                                                    [ 87%]
test/transformers/test_swiglu.py ........                                                                      [ 94%]
test/transformers/test_trainer_integration.py ...                                                              [ 97%]
test/transformers/test_transformers_monkey_patch.py .                                                          [ 98%]
test/triton/test_triton_monkey_patch.py ..                                                                     [100%]

================================================ 114 passed in 59.82s ================================================
```

```zsh
❯ make checkstyle 
flake8 .; flake8_status=$?; \
isort .; isort_status=$?; \
black .; black_status=$?; \
if [ $flake8_status -ne 0 ] || [ $isort_status -ne 0 ] || [ $black_status -ne 0 ]; then \
        exit 1; \
fi
Skipped 1 files
All done! ✨ 🍰 ✨
51 files left unchanged.
```

```zsh
❯ make test-convergence 
HF_DATASETS_OFFLINE=1 pytest --disable-warnings test/convergence
================================================ test session starts =================================================
platform linux -- Python 3.10.12, pytest-8.3.2, pluggy-1.5.0
rootdir: /data/ovis_915/bell/Liger-Kernel.git
plugins: anyio-4.4.0
collected 6 items                                                                                                    

test/convergence/test_mini_models.py ....                                                                      [ 66%]
test/convergence/test_mini_models_no_logits.py ..                                                              [100%]

================================================= 6 passed in 46.98s =================================================
```

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -40,6 +40,8 @@
             "isort>=5.13.2",
             "pre-commit>=3.7.1",
             "torch-tb-profiler>=0.4.1",
+            "pytest>=7.1.2",
```
Comment (ByronHsu) [inline setup.py:43]: awesome! we can also remove `pre-commit` and `torch-tb-profiler`. I forgot to do so. Otherwise LGTM!

**Comment by chiwanpark** [comment]
@ByronHsu Thanks for your review. I've removed the packages from dev dependencies.

**Review by ByronHsu — APPROVED** [review]

**Comment by ByronHsu** [comment]
btw are you in our discord channel? actively looking for contributors https://discord.gg/CX2YmNmn

**Comment by chiwanpark** [comment]
@ByronHsu I just joined the discord channel. :)

**Pull request merged**